pkg/ocm/ocm_test.go

```diff
@@ -0,0 +1,147 @@
+package ocm_test
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+	mocks "github.com/openshift/configuration-anomaly-detection/pkg/ocm/mock"
+)
+
+var _ = Describe("OCM", func() {
+	var (
+		mockCtrl          *gomock.Controller
+		client            *ocm.OcmClient
+		mocOCMConnection  *mocks.MockocmHandlerIf
+		clusterDeployment *hivev1.ClusterDeployment
+		awsAccountClaim   *awsv1alpha1.AccountClaim
+		err               error
+		clustername       string
+		supportRoleArn    string
+	)
+	Describe("When fetching a ClusterDeployment", func() {
+		BeforeEach(func() {
+			mockCtrl = gomock.NewController(GinkgoT())
+			mocOCMConnection = mocks.NewMockocmHandlerIf(mockCtrl)
+			client = &ocm.OcmClient{Comm: mocOCMConnection}
+			err = fmt.Errorf("some error")
+			clustername = "test-cluster"
+			clusterDeployment = &hivev1.ClusterDeployment{Spec: hivev1.ClusterDeploymentSpec{ClusterName: clustername}}
+		})
+		When("cluster ID is valid", func() {
+			It("should return a valid ClusterDeployment", func() {
+				cdstring, _ := json.Marshal(clusterDeployment)
+				mocOCMConnection.EXPECT().OcmGetResourceLive(clustername, "cluster_deployment").Return(string(cdstring), nil).Times(1)
+				cd_out, err := client.GetClusterDeployment(clustername)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(clusterDeployment).Should(Equal(cd_out))
+			})
+		})
+		When("the cluster doesn't exist or any other error happends in the sdk", func() {
+			It("wrapps the error message and returns nil", func() {
+				mocOCMConnection.EXPECT().OcmGetResourceLive(clustername, "cluster_deployment").Return("", err).Times(1)
+				cd_out, err_out := client.GetClusterDeployment(clustername)
+				Expect(err_out).Should(HaveOccurred())
+				Expect(errors.Unwrap(err_out)).Should(Equal(err))
```

Comment:
yea I know... I am just hesitant with defining errors for pinning lines in tests. This should also be just fine. It basically says the error is going to be bubbled up with whatever error message. So would like to leave it now as it is

Reply:
Defining errors would have the advantage that we could export them. In a library this would make sense, but we are writing a client.. so no idea if we really need exported error variables :thinking:

Reply:
I think errors.Unwrap is just one level down, so it assumes you did: `fmt.Errorf("something: %w", actual_err)` and not
```
fmt.Errorf("other: %w", fmt.Errorf("something: %w", actual_err))
```
which might happpen

Reply:
I think also that adding an error is not too much of an overhead, it makes the test clearer, code more explicit and we "lose" a few lines of error code

Reply:
sorry for the delay, @georgettica. I missed your notification. I will have a look on this, that we can merge this PR ASAP.
Consider it done until tomorrow morning.

Reply:
Yes it was.
Just trying to understand how we should change the test if we are moving away of testing the error that comes back